galera/src/ist.cpp

```diff
@@ -204,9 +216,60 @@ IST_determine_recv_addr (gu::Config& conf)
         recv_addr += ":" + gu::to_string(port);
     }
 
+    log_info << "IST receiver addr using " << recv_addr;
     return recv_addr;
 }
 
+static std::string
+IST_determine_recv_bind(gu::Config& conf)
+{
+	std::string recv_bind;
+
+	recv_bind = conf.get(galera::ist::Receiver::RECV_BIND);
+
+	/* check if explicit scheme is present */
+	if (recv_bind.find("://") == std::string::npos) {
+		bool ssl(false);
+
+		try {
+			std::string ssl_key = conf.get(gu::conf::ssl_key);
+			if (ssl_key.length() != 0)
+				ssl = true;
+		} catch (gu::NotSet&) {
+		}
+
+		if (ssl)
+			recv_bind.insert(0, "ssl://");
+		else
+			recv_bind.insert(0, "tcp://");
+	}
+
+	gu::URI rb_uri(recv_bind);
+
+	try /* check for explicit port,
+	 TODO: make it possible to use any free port (explicit 0?) */
+	{
+		rb_uri.get_port();
+	} catch (gu::NotSet&) /* use gmcast listen port + 1 */
+	{
+		int port(0);
+
+		try {
+			port = gu::from_string<uint16_t>(conf.get(galera::BASE_PORT_KEY));
+
+		} catch (...) {
+			port = gu::from_string<uint16_t>(galera::BASE_PORT_DEFAULT);
+		}
+
+		port += 1;
+
+		recv_bind += ":" + gu::to_string(port);
+	}
+
+	log_info<< "IST receiver bind using " << recv_bind;
+	return recv_bind;
+}
+
```

Comment:
Syntactic issue: This code block seems to use tabs for indentation instead of 4 spaces what the rest of Galera code uses.